libxbrzscale.cpp

```diff
@@ -134,8 +138,30 @@ void displayImage(SDL_Surface* surface, const char* message) {
 }
 */
 
+struct imgUint32{
+  uint32_t *data;
+  int iW;
+  int iH;
+};
+std::vector<imgUint32> vImgUint32Cache;
+imgUint32 ImgUint32Cache(int iW,int iH){ //as that memory deletion wont happen promptly, it may consume too much memory too fast unnecessarily becoming a bottle neck on performance
+  for(int i=0;i<vImgUint32Cache.size();i++){
```

Comment:
If you drop W & H parameters for size, you could use bigger buffers for smaller images. This could save some memory and speed things up, as a bigger buffer could be used for smaller images too.

Test loop could be as simple as `if (vImgUint32Cache[i].size >= imageSize)`